tests/test_sails.py

```diff
@@ -0,0 +1,161 @@
+import pytest
+import requests
+import json
+
+"""
+TEST RESPONSES
+Includes all status code responses for both get and post requests.
+"""
+
+def test_get_no_param_200_code():
+  """
+  SUMMARY:  Test that GET method returns a 200 code ('ok')
+  METHOD:   Send a GET request with no parameteres.
+  FAIL:     Status code is not 200.
+  """
+  
+  # Act
+  r = requests.get('http://capstonedd.cs.pdx.edu:1337/users')
+
+  # Assert
+  assert r.status_code == 200
+  
+  
+def test_post_returns_200_code():
+  """
+  SUMMARY:  Test that valid POST body returns a 200 code ('ok')
+  METHOD:   Send a POST request with proper post body.
+  FAIL:     Status code is not 200.
+  """
+  # Arrange
+  payload = {
+    'firstName' : 'Baiyu',
+    'lastName' : 'Liu',
+    'email' : 'baiyu@pdx.edu',
+    'type' : 'buyer'
+  }
+  
+  # Act
+  r = requests.post('http://capstonedd.cs.pdx.edu:1337/users', data=json.dumps(payload))
+
+  # Assert
+  assert r.status_code == 200
+  
+  
+def test_get_param_200_code():
+  """
+  SUMMARY:  Test that GET method returns a 200 code ('ok')
+  METHOD:   Send a GET request with required parameters.
+  FAIL:     Status code is not 200.
+  """
+  # Arrange
+  payload = {
+    'firstName' : 'Baiyu',
+    'lastName' : 'Liu',
+    'email' : 'baiyu@pdx.edu',
+    'type' : 'buyer'
+  }
+  
+  # Act
+  r = requests.get('http://capstonedd.cs.pdx.edu:1337/users', data=json.dumps(payload))
+  
+  # Assert
+  assert r.status_code == 200
+  
+def test_bad_post1_returns_400_code():
+  """
+  SUMMARY:  Test that invalid POST method returns a 400 code ('bad request')
+  METHOD:   (1) Create a improper payload (same email).
+            (2) Send a POST request with improper post body.
+  FAIL:     Status code is not 400.
+  """
+  # Arrange
+  payload = {
+    'firstName' : 'Baiyu',
+    'lastName' : 'Liu',
+    'email' : 'baiyu@pdx.edu',
+    'type' : 'buyer'
+  }
+  
+  # Act
+  r = requests.post('http://capstonedd.cs.pdx.edu:1337/users', data=json.dumps(payload))
+
+  # Assert
+  assert r.status_code == 400
+  
+def test_bad_post2_returns_400_code():
+  """
+  SUMMARY:  Test that invalid POST method returns a 400 code ('bad request')
+  METHOD:   (1) Create a improper payload (no firstName).
+            (2) Send a POST request with improper post body.
+  FAIL:     Status code is not 400.
+  """
+  # Arrange
+  payload = {
+    'lastName' : 'Liu',
+    'email' : 'baiyu1@pdx.edu',
```

Comment:
Definitely. This is just a very simple test. Sourriya asked me to update this on GitHub in order to test the CI. You can change anything you want without notice. 

Thanks.

Baiyu Liu


Thursday, June 4, 2015, 7:02 PM -0700 from Aaron Hampshire  <notifications@github.com>:
>In  tests/test_sails.py :
>> +  r = requests.post('http://capstonedd.cs.pdx.edu:1337/users', data=json.dumps(payload))
> +
> +  # Assert
> +  assert r.status_code == 400
> +  
> +def test_bad_post2_returns_400_code():
> +  """
> +  SUMMARY:  Test that invalid POST method returns a 400 code ('bad request')
> +  METHOD:   (1) Create a improper payload (no firstName).
> +            (2) Send a POST request with improper post body.
> +  FAIL:     Status code is not 400.
> +  """
> +  # Arrange
> +  payload = {
> +    'lastName' : 'Liu',
> +    'email' : 'baiyu1@pdx.edu',
>There should be a better way to handle unique email addresses. This way, we can run the test once, but we can't run it again a second time.
>—
>Reply to this email directly or  view it on GitHub .